les/clientpool.go

```diff
@@ -378,6 +374,18 @@ func (f *clientPool) setLimits(count int, totalCap uint64) {
 	})
 }
 
+// requestCost feeds request cost after serving a request from the given peer.
+func (f *clientPool) requestCost(p *peer, cost uint64) {
```

Comment:
Why did you remove the `balanceTracker` link from `peer` and use this function instead? We can do this if there is a good reason, I'm just curious why you needed this extra function and extra map lookup every time the stats are updated.

Reply:
I think balanceTracker is just to collect stat for clientpool, so other components other than clientpool should not know and maintain these things.

Although we introduce another map lookup, it's really trivial. I think it's easier to maintain.